app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardVMStep/PlanWizardVMStep.js

```diff
@@ -124,9 +124,13 @@ class PlanWizardVMStep extends React.Component {
     } else if (!isValidatingVms && validationServiceCalled) {
       // set make rows editable so they can be selected
       const validVms = Immutable.asMutable(valid_vms, { deep: true });
-      const inValidsVms = Immutable.asMutable(invalid_vms, { deep: true });
+      const inValidsVms = Immutable.asMutable(invalid_vms, { deep: true }).concat(
+        validVms.filter(vm => vm.valid === false)
```

Comment:
If there are VMs returned in `valid_vms` with `vm.valid === false`, won't this cause them to appear as duplicates in `combined` below?

Reply:
`valid` VMs and `invalid` VMs are two distinct sets that we get from the backend.

The UI vets the `valid` VMs for duplicates and marks the duplicate valid rows as invalid.
So in the above code I'm just ensuring that we append those duplicate valid rows to the invalid array, so that these VMs would be selectable.

So to answer you question, by doing the above, we won't see duplicates :)

Reply:
Makes sense! Thanks.